src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IfStatement.cs

```diff
@@ -59,7 +59,7 @@ public override BoundNode VisitIfStatement(BoundIfStatement node)
 
                 builder.Add(new BoundConditionalGoto(rewrittenCondition.Syntax, rewrittenCondition, false, afterif));
                 builder.Add(rewrittenConsequence);
-                builder.Add(new BoundLabelStatement(syntax, afterif));
+                builder.Add(new BoundSequencePoint(syntax: null, statementOpt: new BoundLabelStatement(syntax, afterif)));
```

Comment:
It seems like   while/for loops  with a single statement body could also be affected.